### PR TITLE
Update to gradle 7.0

### PR DESCRIPTION
**Pull request by dhardy**
**WIP**

This updates Gradle and several dependencies to the latest versions. I don't know if this is acceptable or if there are compatibility requirements on using older versions.

It's also broken. I'm not very familiar with gradle and could use some help here:
```
$ ./gradlew check

> Configure project :
Platform is: linux

> Task :downloadIceAdapter UP-TO-DATE
Download https://github.com/FAForever/java-ice-adapter/releases/download/v3.0.0/faf-ice-adapter.jar

> Task :downloadUnixUid
Download https://github.com/FAForever/uid/releases/download/v4.0.4/faf-uid

> Task :downloadWindowsUid UP-TO-DATE
Download https://github.com/FAForever/uid/releases/download/v4.0.4/faf-uid.exe

> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Some problems were found with the configuration of task ':compileJava' (type 'JavaCompile').
  - Type 'JavaCompile' property 'options.compilerArgumentProviders.apt$0.name' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'JavaCompile' property 'options.compilerArgumentProviders.apt$0.publicType' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/7.0/userguide/command_line_interface.html#sec:command_line_warnings
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,16 +10,15 @@ buildscript {
...
   dependencies {
     classpath("gradle.plugin.install4j.install4j:gradle_plugin:7.0.7")
-    classpath("org.springframework.build.gradle:propdeps-plugin:${propdepsVersion}")
     classpath "org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}"
   }
 }
 
 plugins {
-  id "com.install4j.gradle" version "7.0.7"
+  id "com.install4j.gradle" version "9.0.2"
```
Comment (Brutus5000) [inline build.gradle]: This will most probably break. Should only be changed by the maintainer.

File: build.gradle [hunk 2]
```diff
@@ -205,68 +211,68 @@ repositories {
...
+  compileOnly 'junit:junit:4.13'
+  compileOnly("org.testfx:testfx-core:4.0.8-alpha") {
     exclude module: 'guava'
   }
-  testCompile("org.testfx:testfx-junit:4.0.8-alpha")
-  testCompile("com.natpryce.hamcrest:hamcrest-reflection:0.1-2")
-  testCompile("org.springframework.boot:spring-boot-starter-test")
+  compileOnly("org.testfx:testfx-junit:4.0.8-alpha")
+  compileOnly("com.natpryce.hamcrest:hamcrest-reflection:0.1-2")
+  compileOnly("org.springframework.boot:spring-boot-starter-test")
```
Comment (Brutus5000) [inline build.gradle]: `testCompileOnly`

File: gradle.properties [hunk 3]
```diff
@@ -2,17 +2,16 @@ version=unspecified
...
-propdepsVersion=0.0.7
-springBootVersion=2.4.0
+springBootVersion=2.4.5
 springSecurityOauth2AutoconfigureVersion=2.3.4.RELEASE
 fafCommonsVersion=2474a361d0534ae77c5fefe9874d4134b8156c94
 qBuildersVersion=1.6
 jSkillsVersion=8b333ec63d
 jsonapiConverterVersion=0.10
-install4jRuntimeVersion=7.0.7
+install4jRuntimeVersion=9.0.2
```
Comment (Brutus5000) [inline gradle.properties]: This will most probably break. Should only be changed by the maintainer.

File: build.gradle [hunk 4]
```diff
@@ -70,6 +68,14 @@ processResources {
...
 }
 
+configurations {
+    provided
+}
+
+sourceSets {
+    main { compileClasspath += configurations.provided }
+}
+
```
Comment (Brutus5000) [inline build.gradle:78]: Why is this required? Any docs mentioning this? Looks pretty standard.

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 4 above)
Comment (dhardy) [reply build.gradle:78]: It came from here and replaces `org.springframework.build.gradle:propdeps-plugin` (which I had problems with).

Edit: I mean [here](https://stackoverflow.com/questions/30731084/provided-dependency-in-gradle).

**Comment by dhardy** [comment]
Updated; same error. Is it from our `build.gradle` or a plugin?

File: build.gradle (hunk 4 above)
Comment (dev-nulled) [reply build.gradle:78]: Still working through the updates at work... but noticed the gradle 7 support in Springboot 2.5.0.

Comment (Brutus5000) [reply build.gradle:78]: The code above is not needed. The whole provided scope is a maven thing. We can just use implemenation for the install4j dependency.
Basically we solved all issues here except for 1 issue with install4j that we can't fix ourselves on building a release.


https://github.com/FAForever/downlords-faf-client/commits/feature/gradle7

Maybe updating to install4j 9 can fix it. Otherwise we need to raise a bugticket.

**Comment by Sheikah45** [comment]
Superseded by #2243

**Comment by dhardy** [comment]
Actually, #2250. Great!
